iree/hal/cuda/dynamic_symbols.c

```diff
@@ -53,6 +74,16 @@ iree_status_t iree_hal_cuda_dynamic_symbols_initialize(
         IREE_STATUS_UNAVAILABLE,
         "CUDA runtime library not available; ensure installed and on path");
   }
+  status = iree_dynamic_library_load_from_files(
+      IREE_ARRAYSIZE(kCUPTILoaderSearchNames), kCUPTILoaderSearchNames,
+      IREE_DYNAMIC_LIBRARY_FLAG_NONE, allocator, &out_syms->cupti_library);
+  if (iree_status_is_not_found(status)) {
+    iree_status_ignore(status);
```

Comment:
we don't want to require cupti - if the library fails to load you need to continue without it

Reply:
How should I handle the `status`? If tracing is enabled but CUPTI fails to load, I expect these symbols to fail or remain null. 

I'd need a way to gracefully communicate this to the tracing library (and potentially the user) for proper handling. `dynbamic_cupti_test.cc` relies on the status to tell if all of the libraries are loaded.

Reply:
You could only return an error if `IREE_ENABLE_CUPTI` is defined, or have any code paths that use these symbols no-op if their symbols were not found.